Nim/Nim/Nim.cpp

```diff
@@ -0,0 +1,133 @@
+using namespace std;
+
+#include <iostream>
+#include <string>
+#include <cmath>
+
+int MatchSelect(int Num, int Total, string PlayerName) // I always get the total matches left for other player from here.
+{
+    if (Num > 0 && Num < 4)
+    {
+        Total -= Num;
+    }
+    else
+    {
+        cout << "Invalid Number, you missed your chance to decide how many matches to remove from the table" << endl;
+        Num = rand() % 3 + 1;
```

Comment:
Here, you could have created a method for rolling a random number `int GetRandomNumber(int min, int max);`